applications/tari_merge_mining_proxy/src/block_template_data.rs

```diff
@@ -37,18 +60,38 @@ impl BlockTemplateRepository {
     }
 
     pub async fn get<T: AsRef<[u8]>>(&self, hash: T) -> Option<BlockTemplateData> {
+        debug!(
+            target: LOG_TARGET,
+            "Retrieving blocktemplate with merge mining hash: {:?}",
+            hash.as_ref()
+        );
         let b = self.blocks.read().await;
-        b.get(hash.as_ref()).cloned()
+        match b.get(hash.as_ref()).cloned() {
+            Some(item) => Some(item.data),
+            None => None,
+        }
     }
 
     pub async fn save<T: AsRef<[u8]>>(&mut self, hash: T, block_template: BlockTemplateData) {
+        debug!(
+            target: LOG_TARGET,
+            "Saving blocktemplate with merge mining hash: {:?}",
+            Vec::from(hash.as_ref())
+        );
         let mut b = self.blocks.write().await;
-        b.insert(Vec::from(hash.as_ref()), block_template);
+        let repository_item = BlockTemplateRepositoryItem::new(block_template);
+        b.insert(Vec::from(hash.as_ref()), repository_item);
     }
 
-    pub async fn remove<T: AsRef<[u8]>>(&mut self, hash: T) -> Option<BlockTemplateData> {
+    pub async fn remove_outdated(&mut self) {
+        debug!(target: LOG_TARGET, "Removing outdated blocktemplates");
         let mut b = self.blocks.write().await;
-        b.remove(hash.as_ref())
+        let threshold = Utc::now() - Duration::minutes(20);
+        for item in b.clone().iter() {
+            if item.1.datetime() < threshold {
+                b.remove(item.0);
```

Comment:
Could we print the ones that are removed as well?

Reply:
Will add it